WooCommerce/Classes/ViewRelated/BottomSheet/ListSelector/BottomSheetListSelectorDataSource.swift

```diff
@@ -0,0 +1,23 @@
+import UIKit
+
+/// A generic data source for the bottom sheet list selector UI `BottomSheetListSelectorViewController`.
+///
+protocol BottomSheetListSelectorDataSource {
+    associatedtype Model: Equatable
+    associatedtype Cell: UITableViewCell
+
+    /// A list of models to render the list.
+    var data: [Model] { get }
+
+    /// The model that is currently selected in the list.
+    var selected: Model? { get }
+
+    /// Called when a different model is selected.
+    mutating func handleSelectedChange(selected: Model)
```

Comment:
yea I agree this isn't strictly a data source since `*DataSource` could sound confusing with `UITableViewDataSource`, and it might be a bit over complicated with a separate `*Delegate`. `*Command` sounds great, we also have a few `*Command`s in the codebase. will update!

Reply:
updated in: https://github.com/woocommerce/woocommerce-ios/pull/2155/commits/ebd126d281030f8cb0edc5e5be07d83078559ad1 and https://github.com/woocommerce/woocommerce-ios/pull/2155/commits/fafca4dc65ff44274d45950cf4287d4e3062a4f8